lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js

```diff
@@ -0,0 +1,144 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The original C code, long comment, copyright, license, and constants are from [Cephes]{@link http://www.netlib.org/cephes}. The implementation follows the original, but has been modified for JavaScript.
+*
+* ```text
+* Copyright 1984, 1995, 2000 by Stephen L. Moshier
+*
+* Some software in this archive may be from the book _Methods and Programs for Mathematical Functions_ (Prentice-Hall or Simon & Schuster International, 1989) or from the Cephes Mathematical Library, a commercial product. In either event, it is copyrighted by the author. What you see here may be used freely but it comes with no support or guarantee.
+*
+* Stephen L. Moshier
+* moshier@na-net.ornl.gov
+* ```
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
+var sqrtf = require( '@stdlib/math/base/special/sqrtf' );
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+var PI02F = require( '@stdlib/constants/float32/half-pi' );
+var polyp = require( './poly_p.js' );
+
+
+// MAIN //
+
+/**
+* Computes the arcsine of a single-precision floating-point number.
+*
+* ## Method
+*
+* -   A rational function of the form
+*
+*     ```tex
+*     x + x^3 \frac{P(x^2)}{Q(x^2)}
+*     ```
+*
+*     is used for \\(\|x\|\\) in the interval \\(\[0, 0.5\]\\). If \\(\|x\| > 0.5\\), it is transformed by the identity
+*
+*     ```tex
+*     \operatorname{asinf}(x) = \frac{\pi}{2} - 2 \operatorname{asinf}( \sqrt{ (1-x)/2 } )
+*     ```
+*
+* ## Notes
+*
+* -   Relative error:
+*
+*     | arithmetic | domain | # trials | peak    | rms     |
+*     |:-----------|:-------|:---------|:--------|:--------|
+*     | IEEE       | -1, 1  | 10^5     | 1.4e-7  | 4.2e-8  |
```

Comment:
```suggestion
*     | IEEE       | -1, 1  | 10^5     | 2.5e-7  | 5.0e-8  |
```

Taken from the Cephes source.